vyper/builtin_functions/functions.py

```diff
@@ -1006,12 +1069,19 @@ def evaluate(self, node):
         return vy_ast.Int.from_node(node, value=int(value * denom))
 
     def fetch_call_return(self, node):
-        validate_expected_type(node.args[0], NumericAbstractType())
+        self.infer_arg_types(node)
         return self._return_type
 
+    def infer_arg_types(self, node):
+        validate_expected_type(node.args[0], NumericAbstractType())
+        value_type = get_possible_types_from_node(node.args[0]).pop()
+        denomination_type = get_exact_type_from_node(node.args[1])
```

Comment:
this is probably a good place to check that the value of `node.args[1]` is a valid denomination

Reply:
I think this is already checked in `evaluate`.

Reply:
Move to new helper and throw exception or return None